lib/matplotlib/axes/_axes.py

```diff
@@ -1997,20 +1998,14 @@ def bar(self, left, height, width=0.8, bottom=None, **kwargs):
         label = kwargs.pop('label', '')
         tick_labels = kwargs.pop('tick_label', None)
 
-        def make_iterable(x):
-            if not iterable(x):
-                return [x]
-            else:
-                return x
-
         # make them safe to take len() of
         _left = left
-        left = make_iterable(left)
```

Comment:
Edit: Actually asarray doesn't even copy the underlying buffer for pandas series:
```
In [1]: s = pd.Series([1, 2, 3]); t = np.asarray(s); t[0] = 42; s
Out[1]: 
0    42
1     2
2     3
dtype: int64
```
so the point is moot.